docs/src/reference/compilation.rst

```diff
@@ -316,6 +316,128 @@ customize your build.
+        name = "hello",
+        ext_modules = cythonize("lib.pyx"),
+    )
+
+Here are the options that are available:
+
+``docstrings`` (True / False)
+    Whether or not to include docstring in the Python extension. If False, the binary size
+    will be smaller, but the ``__doc__`` attribute of any class or function will be an
+    empty string.
+    Default is True.
+
+``embed_pos_in_docstring`` (True / False)
+    Embed the source code position in the docstrings of functions and classes.
+    Default is False.
+
+``emit_code_comments`` (True / False)
+    Copy the original source code line by line into C code comments
+    in the generated code file to help with understanding the output.
+    Default is True.
+
+``generate_cleanup_code`` (integer)
+    Decref global variables in this module on exit for garbage collection.
+    0: None, 1+: interned objects, 2+: cdef globals, 3+: types objects
+    Mostly for reducing noise in Valgrind, only executes at process exit
+    (when all memory will be reclaimed anyways).
+    Default is 0.
+
+``annotate`` (True / False)
+    Generate an annotated HTML version of the input source files.
+    This has the same effect as the ``annotate`` argument in :func:`cythonize`.
+    Default is False.
+
+``fast_fail`` (True / False)
+    This will abort the compilation on the first error occurred rather than trying
+    to keep going and printing further error messages.
+    Default is False.
+
+``warning_errors`` (True / False)
+    Make all warnings into errors.
+    Default is False.
+
+``error_on_unknown_names`` (True / False)
+    Make unknown names an error.  Python raises a NameError when
+    encountering unknown names at runtime, whereas this option makes
+    them a compile time error.  If you want full Python compatibility,
+    you should disable this option and also 'cache_builtins'.
+    Default is True.
+
+``error_on_uninitialized`` (True / False)
+    Make uninitialized local variable reference a compile time error.
+    Python raises UnboundLocalError at runtime, whereas this option makes
+    them a compile time error. Note that this option affects only variables
+    of "python object" type.
+    Default is True.
+
+``convert_range`` (True / False)
+    This will convert statements of the form ``for i in range(...)``
+    to ``for i from ...`` when ``i`` is a C integer type, and the direction
+    (i.e. sign of step) can be determined.
+    WARNING: This may change the semantics if the range causes assignment to
+    i to overflow. Specifically, if this option is set, an error will be
+    raised before the loop is entered, whereas without this option the loop
+    will execute until an overflowing value is encountered.
+    Default is True.
+
+``cache_builtins`` (True / False)
+    Perform lookups on builtin names only once, at module initialisation
+    time.  This will prevent the module from getting imported if a
+    builtin name that it uses cannot be found during initialisation.
```

Comment:
I'd add a comment here:
"Note that some legacy builtins are automatically remapped from their Python 2 names to their Python 3 names by Cython when building in Python 3.x, so that they do not get in the way even if this option is enabled."